apps/site/lib/site_web/controllers/trip_plan_controller.ex

```diff
@@ -56,6 +65,37 @@ defmodule SiteWeb.TripPlanController do
     assign(conn, :requires_google_maps?, true)
   end
 
+  @spec with_fares([Itinerary.t()]) :: [Itinerary.t()]
+  defp with_fares(itineraries) do
+    Enum.map(itineraries, fn itinerary ->
+      legs_with_fares = itinerary.legs |> Enum.map(&leg_with_fares/1)
+
+      %{itinerary | legs: legs_with_fares}
+    end)
+  end
+
+  @spec leg_with_fares(Leg.t()) :: Leg.t()
+  defp leg_with_fares(%Leg{mode: %PersonalDetail{}} = leg) do
+    leg
+  end
+
+  defp leg_with_fares(%Leg{mode: %TransitDetail{}} = leg) do
+    route = Routes.Repo.get(leg.mode.route_id)
+    trip = Schedules.Repo.trip(leg.mode.trip_id)
+    origin_id = leg.from.stop_id
+    destination_id = leg.to.stop_id
+    lowest_fare = Fares.MinMaxFare.lowest_fare(route, trip, origin_id, destination_id)
+    highest_fare = Fares.MinMaxFare.highest_fare(route, trip, origin_id, destination_id)
```

Comment:
It looks great to me :smiley:
The only 'nitpicky' thing I was wondering was if the `MinMaxFare` file I created early on should be renamed to something else for consistency because in it `min` and `max` refer to `highest` and `lowest`. But this is not really a big deal.

Reply:
👍 sure, that's a good idea.